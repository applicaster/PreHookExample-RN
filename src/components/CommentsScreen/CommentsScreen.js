import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {
  FlatList,
  Modal,
  View,
  ScrollView,
  Text,
} from 'react-native';
import EventContainer from '../EventContainer';
import CommentContainer from '../CommentContainer';
import MediaDetailsModal from '../MediaDetailsModal';
import BackButton from '../BackButton';

class CommentsScreen extends Component {
  static navigationOptions = ({ navigation, screenProps }) => ({
    ...screenProps,
    headerTitle: "Comments",
    headerLeft: <BackButton onPress={() => {navigation.goBack(); }} />,
  });

  constructor(props) {
    super(props);
    this.renderItem = this.renderItem.bind(this);
  }

  renderItem(data) {
    const { event } = this.props;
    const { item } = data;
    if (item.id === event.id) {
      return <EventContainer key={event.id} event={event} />;
    }
    
    return <CommentContainer key={item.id} comment={item} />;
  }

  render() {
    const { event, isMediaModalVisible, toggleModal } = this.props;
    const backgroundFeedColor = { backgroundColor: this.context.backgroundColor };
    const comments = [event].concat(event.comments);;
  
    return (<View style={[backgroundFeedColor]}>
        <FlatList
          data={comments}
          renderItem={this.renderItem}
          keyExtractor={(item) => item.id }
          style={[backgroundFeedColor]} contentContainerStyle={[backgroundFeedColor]}
          initialNumToRender={5}
          onEndReached={() => {}}
          onEndReachedThreshold={2}
        />
        <MediaDetailsModal />
    </View>
        
    );
  }
}

CommentsScreen.propTypes = {
  event: PropTypes.object,
  navigation: PropTypes.object,
};

CommentsScreen.contextTypes = {
  backgroundColor: PropTypes.string,
};

export default CommentsScreen;
