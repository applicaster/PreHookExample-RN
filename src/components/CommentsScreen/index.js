import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { toggleModal } from '../../actions';

import {
  getActiveEvent,
  getMediaModalVisibility,
} from '../../selectors';

import CommentsScreen from './CommentsScreen';

const mapStateToProps = state => ({
  event: getActiveEvent(state),
  isMediaModalVisible: getMediaModalVisibility(state),
});

const mapDispatchToProps = dispatch => bindActionCreators({
  toggleModal,
}, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(CommentsScreen);
