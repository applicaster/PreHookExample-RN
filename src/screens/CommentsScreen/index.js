import { connect } from 'react-redux';
import {
  getActiveEvent,
} from '../../selectors';

import CommentsScreen from './CommentsScreen';

const mapStateToProps = state => ({
  event: getActiveEvent(state),
});

export default connect(mapStateToProps)(CommentsScreen);
