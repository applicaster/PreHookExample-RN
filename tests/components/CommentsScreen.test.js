import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { event, comment } from './testDataUtils';
import CommentsScreen from '../../src/components/CommentsScreen/CommentsScreen';

jest.mock('../../src/components/EventContainer', () => 'EventContainer');
jest.mock('../../src/components/ModalScreen', () => 'ModalScreen');

const props = () => ({
  event: event('id'),
  navigation: {},
});

describe('CommentsScreen', () => {
  test('component renders correctly with no comments', () => {
    const tree = renderer.create(
      <CommentsScreen
        {...props()}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('component renders correctly with comments', () => {
    const tree = renderer.create(
      <CommentsScreen
        {...props(1, 'image', 'instagram', 'applicaster', [comment(1), comment(2)])}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
