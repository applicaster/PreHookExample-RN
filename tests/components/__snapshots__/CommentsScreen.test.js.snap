// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentsScreen component renders correctly with comments 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": undefined,
      },
    ]
  }
>
  <RCTScrollView
    contentContainerStyle={
      Array [
        Object {
          "backgroundColor": undefined,
        },
      ]
    }
    data={
      Array [
        Object {
          "caption": "Low. Tarmac T2 , CasCam, SPC LCAs.
      Thanks @azpinstalls for the work, parts, advice and setup.
      #brz #gt86 #ft86",
          "comments": Array [],
          "commentsCount": 0,
          "createdAt": 1501886235,
          "id": "id",
          "images": Object {
            "default": Object {
              "height": 504,
              "url": "https://scontent.cdninstagram.com/t51.2885-15/s640x640/sh0.08/e35/20590092_108583273161709_7296810043293302784_n.jpg",
              "width": 640,
            },
            "high_resolution": Object {
              "height": 252,
              "url": "https://scontent.cdninstagram.com/t51.2885-15/s320x320/e35/20590092_108583273161709_7296810043293302784_n.jpg",
              "width": 320,
            },
          },
          "likesCount": 30,
          "source": "instagram",
          "type": "image",
          "user": Object {
            "avatarImageUrl": "https://scontent.cdninstagram.com/t51.2885-19/11356956_452315591560419_1118499770_a.jpg",
            "id": "220971565",
            "name": "Test Name",
            "userName": "applicaster",
          },
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={5}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReached={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Array [
        Object {
          "backgroundColor": undefined,
        },
      ]
    }
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <EventContainer
          event={
            Object {
              "caption": "Low. Tarmac T2 , CasCam, SPC LCAs.
            Thanks @azpinstalls for the work, parts, advice and setup.
            #brz #gt86 #ft86",
              "comments": Array [],
              "commentsCount": 0,
              "createdAt": 1501886235,
              "id": "id",
              "images": Object {
                "default": Object {
                  "height": 504,
                  "url": "https://scontent.cdninstagram.com/t51.2885-15/s640x640/sh0.08/e35/20590092_108583273161709_7296810043293302784_n.jpg",
                  "width": 640,
                },
                "high_resolution": Object {
                  "height": 252,
                  "url": "https://scontent.cdninstagram.com/t51.2885-15/s320x320/e35/20590092_108583273161709_7296810043293302784_n.jpg",
                  "width": 320,
                },
              },
              "likesCount": 30,
              "source": "instagram",
              "type": "image",
              "user": Object {
                "avatarImageUrl": "https://scontent.cdninstagram.com/t51.2885-19/11356956_452315591560419_1118499770_a.jpg",
                "id": "220971565",
                "name": "Test Name",
                "userName": "applicaster",
              },
            }
          }
        />
      </View>
    </View>
  </RCTScrollView>
  <ModalScreen />
</View>
`;

exports[`CommentsScreen component renders correctly with no comments 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": undefined,
      },
    ]
  }
>
  <RCTScrollView
    contentContainerStyle={
      Array [
        Object {
          "backgroundColor": undefined,
        },
      ]
    }
    data={
      Array [
        Object {
          "caption": "Low. Tarmac T2 , CasCam, SPC LCAs.
      Thanks @azpinstalls for the work, parts, advice and setup.
      #brz #gt86 #ft86",
          "comments": Array [],
          "commentsCount": 0,
          "createdAt": 1501886235,
          "id": "id",
          "images": Object {
            "default": Object {
              "height": 504,
              "url": "https://scontent.cdninstagram.com/t51.2885-15/s640x640/sh0.08/e35/20590092_108583273161709_7296810043293302784_n.jpg",
              "width": 640,
            },
            "high_resolution": Object {
              "height": 252,
              "url": "https://scontent.cdninstagram.com/t51.2885-15/s320x320/e35/20590092_108583273161709_7296810043293302784_n.jpg",
              "width": 320,
            },
          },
          "likesCount": 30,
          "source": "instagram",
          "type": "image",
          "user": Object {
            "avatarImageUrl": "https://scontent.cdninstagram.com/t51.2885-19/11356956_452315591560419_1118499770_a.jpg",
            "id": "220971565",
            "name": "Test Name",
            "userName": "applicaster",
          },
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={5}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReached={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Array [
        Object {
          "backgroundColor": undefined,
        },
      ]
    }
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <EventContainer
          event={
            Object {
              "caption": "Low. Tarmac T2 , CasCam, SPC LCAs.
            Thanks @azpinstalls for the work, parts, advice and setup.
            #brz #gt86 #ft86",
              "comments": Array [],
              "commentsCount": 0,
              "createdAt": 1501886235,
              "id": "id",
              "images": Object {
                "default": Object {
                  "height": 504,
                  "url": "https://scontent.cdninstagram.com/t51.2885-15/s640x640/sh0.08/e35/20590092_108583273161709_7296810043293302784_n.jpg",
                  "width": 640,
                },
                "high_resolution": Object {
                  "height": 252,
                  "url": "https://scontent.cdninstagram.com/t51.2885-15/s320x320/e35/20590092_108583273161709_7296810043293302784_n.jpg",
                  "width": 320,
                },
              },
              "likesCount": 30,
              "source": "instagram",
              "type": "image",
              "user": Object {
                "avatarImageUrl": "https://scontent.cdninstagram.com/t51.2885-19/11356956_452315591560419_1118499770_a.jpg",
                "id": "220971565",
                "name": "Test Name",
                "userName": "applicaster",
              },
            }
          }
        />
      </View>
    </View>
  </RCTScrollView>
  <ModalScreen />
</View>
`;
